app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt

```diff
@@ -86,35 +88,31 @@ class LockboxAutofillService(
             return null
         }
 
-        val dataset = datasetForPossibleValues(possibleValues, parsedStructure)
-        // future parts of this method include adding any authentication steps
+        val builder = FillResponse.Builder()
 
-        return FillResponse.Builder()
-            .addDataset(dataset)
-            .build()
+        possibleValues
+            .map { serverPasswordToDataset(parsedStructure, it) }
+            .map(builder::addDataset)
```

Comment:
Is it ok to use two `maps` concurrently?

Reply:
Yes. It looks a bit weird. Strictly, I should use a `forEach`, but passing `.forEach(builder::addDataset)` didn't compile.

Reply:
Makes sense :)